src/python/gudhi/datasets/remote.py

```diff
@@ -0,0 +1,101 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Hind Montassif
+#
+# Copyright (C) 2021 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import hashlib
+
+from os.path import join, exists
+from os import makedirs
+
+from urllib.request import urlretrieve
+
+
+def _checksum_sha256(file_path):
+    """
+    Compute the file checksum using sha256
+
+    Parameters
+    ----------
+    file_path: string
+        Full path of the created file.
+
+    Returns
+    -------
+        The hex digest of file_path
+    """
+    sha256_hash = hashlib.sha256()
+    chunk_size = 4096
+    with open(file_path,"rb") as f:
+        # Read and update hash string value in blocks of 4K
+        while True:
+            buffer = f.read(chunk_size)
+            if not buffer:
+                break
+            sha256_hash.update(buffer)
+    return sha256_hash.hexdigest()
+
+def fetch(url, filename, dirname = "remote_datasets", file_checksum = None):
```

Comment:
Do we keep `fetch` public, or do we make it private ?

Reply:
I am more in favor of making it private because I find it tedious that the user should go look for the corresponding url, let alone the checksum ... But maybe there are other reasons to keep it public (not to be forced to implement a function for every dataset that we make available for example ?).

Reply:
ok, keep it public.